File: 06-1.rb
#!/usr/bin/env ruby

lines = File.read('06.input').lines.map(&:strip)

times = lines[0].split(':')[1].split.map(&:to_i)
records = lines[1].split(':')[1].split.map(&:to_i)

num = 1
times.each_index do |race|
  time = times[race]
  record = records[race]

  wins = 0
  0.upto(time) do |hold|
    speed = hold
    distance = (time - hold) * speed
    wins += 1 if distance > record
  end
  num *= wins if wins > 0
end

print num, "\n"

File: 06-1.py
#!/usr/bin/env python

lines = [row.strip() for row in open('06.input').readlines()]

times = [int(n) for n in lines[0].split(':')[1].split()]
records = [int(n) for n in lines[1].split(':')[1].split()]

num = 1
for race in range(len(times)):
    time = times[race]
    record = records[race]

    wins = 0
    for hold in range(time + 1):
        speed = hold
        distance = (time - hold) * speed
        if distance > record:
            wins += 1

    if wins > 1:
        num *= wins

print(num)
